compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BEnumSymbol.java

```diff
@@ -53,6 +53,11 @@ public BEnumSymbol(List<BConstantSymbol> members, long flags, Name name, Package
         this(members, flags, name, name, pkgID, type, owner, pos, origin);
     }
 
+    @Override
+    public SymbolKind getKind() {
+        return this.kind != null ? this.kind : super.getKind();
```

Comment:
What's the need for this conditional? Wouldn't simply returning `SymbolKind.ENUM` do?